GVRf/Framework/framework/src/main/java/org/gearvrf/GVRContext.java

```diff
@@ -709,6 +725,13 @@ public void runOnTheFrameworkThread(final Runnable runnable) {
         mHandler.post(runnable);
     }
 
+    /**
+     * Execute a runnable on the dedicated V8 thread
+     */
+    public void runOnV8Thread(final Runnable runnable) {
```

Comment:
Why would we need a public api for that?

Reply:
And answering to it myself - because of the fact x3d is in its own package. I am pretty sure because of incidents like this that x3d should be moved to the gearvrf package. Now we are exposing an internal detail in the most prominent gvrf class just so x3d can get to it.

Reply:
How might this change if this V8thread were moved into Extensions, just like gvr-physics?
If you are suggesting every X3D file be moved from it's own 'X3D' package into GearVR, then I suppose every X3D Java file could be prefixed with "X3D" in its name  just because these files are exclusive to X3D implementation.

Reply:
Since scripting in its own extension is not happening right now, in the meantime we could hide unnecessary public methods if X3D was in the gearvrf package. If one day we clean it all up and everything can happen through real public API then X3D does belong to its own package.